tools/docker/builder/openwrt/Dockerfile

```diff
@@ -0,0 +1,100 @@
+ARG HOME=/home/openwrt
+
+####
+## OpenWrt pre-requisites and "openwrt" user
+####
+FROM ubuntu:18.04 as openwrt-prerequisites
+
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+    build-essential libncurses5 libncurses5-dev python python3 \
+    unzip git ca-certificates gawk wget file bash \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN useradd -ms /bin/bash openwrt
+
+####
+## Pre-built OpenWrt (and extracted sdk)
+####
+FROM openwrt-prerequisites as openwrt-prebuilt
+
+ARG HOME
+COPY --chown=openwrt:openwrt configs/sdk_config $HOME/sdk_config
+USER openwrt
+
+ARG OPENWRT_REPOSITORY=https://git.prpl.dev/prplwrt/prplwrt.git
+ARG OPENWRT_VERSION=9d2efd
+ARG TARGET=mvebu
+
+WORKDIR $HOME
+
+RUN git clone "$OPENWRT_REPOSITORY" openwrt \
+    && cd openwrt \
+    && git checkout "$OPENWRT_VERSION"
+
+WORKDIR $HOME/openwrt
+
+RUN cp "${HOME}/sdk_config" .config \
+    && echo "CONFIG_TARGET_${TARGET}=y" >> .config \
+    && cat .config \
+    && make defconfig \
+    && make -j$(nproc) 2>&1 | tee build.log
+
+RUN TAR="$(find bin/targets -name 'openwrt-sdk*' -exec echo {} \;)" \
+    && mkdir -p "${HOME}/openwrt_sdk" \
+    && mv -v "$TAR" "${HOME}/openwrt_sdk" \
+    && TAR="$(basename $TAR)" \
```

Comment:
Indeed, that's so much simpler. I also skipped the "rm" part entirely, since there is no real reason to delete it (from the `openwrt` dir; it won't be in `openwrt_sdk`).